libcontainer/cgroups/fs/cpuacct.go

```diff
@@ -78,27 +78,41 @@ func getCpuUsageBreakdown(path string) (uint64, uint64, error) {
 	)
 
 	// Expected format:
-	// user <usage in ticks>
-	// system <usage in ticks>
+	// key1 value1
+	// key2 value2
+	// ...
 	data, err := ioutil.ReadFile(filepath.Join(path, cgroupCpuacctStat))
 	if err != nil {
 		return 0, 0, err
 	}
+
 	fields := strings.Fields(string(data))
-	if len(fields) != 4 {
-		return 0, 0, fmt.Errorf("failure - %s is expected to have 4 fields", filepath.Join(path, cgroupCpuacctStat))
-	}
-	if fields[0] != userField {
-		return 0, 0, fmt.Errorf("unexpected field %q in %q, expected %q", fields[0], cgroupCpuacctStat, userField)
+	if len(fields)%2 != 0 {
+		return 0, 0, fmt.Errorf("failure - %s is expected to have even fields, actually %d", filepath.Join(path, cgroupCpuacctStat), len(fields))
 	}
-	if fields[2] != systemField {
-		return 0, 0, fmt.Errorf("unexpected field %q in %q, expected %q", fields[2], cgroupCpuacctStat, systemField)
+
+	statDict := make(map[string]string)
+	for i := 0; i < len(fields); i += 2 {
+		if _, ok := statDict[fields[i]]; ok {
+			return 0, 0, fmt.Errorf("failure - %s has %s appears more than once", filepath.Join(path, cgroupCpuacctStat), fields[i])
```

Comment:
"appears" should probably be "appear";

```suggestion
			return 0, 0, fmt.Errorf("failure - %s has %s appear more than once", filepath.Join(path, cgroupCpuacctStat), fields[i])
```

of perhaps (failure is a bit redundant?);

```suggestion
			return 0, 0, fmt.Errorf("duplicate field %q in %q", fields[i], filepath.Join(path, cgroupCpuacctStat))
```